Create TITAN.s4ext

sindhurathiru opened the pull request:
Thank you for contributing to 3D Slicer!

- **To add a new extension with this pull request**: Please keep content of "TODO list for submitting a new extension" section and put an 'x' character in the brackets for each todo item to indicate that you have accomplished that prerequisite.

- **To update an existing extension with this pull request**: Please delete all text in this template and just describe which extension is updated and optionally tell us in a sentence what has been changed. To make extension updates easier in the future (so that you don't have to submit a new pull request after each change in your extension), you may consider replacing specific git hash in your s4ext file by a branch name (for example: `master` or `release` for nightly version; `(majorVersion).(minorVersion)` such as `4.10` for stable Slicer version).

# TODO list for submitting a new extension

[To make sure users can find your extension, understand what it is intended for and how to use it, please complete the checklist below. You do not need to complete all the item by the time you submit the pull request, but most likely the changes will only be merged if all the tasks are done. See more information about the submission process here: https://slicer.readthedocs.io/en/latest/developer_guide/extensions.html]

- [x] Extension has a reasonable name (not too general, not too narrow, suggests what the extension is for)
- [x] Repository name is Slicer+ExtensionName
- [x] Repository is associated with `3d-slicer-extension` GitHub topic so that it is listed [here](https://github.com/topics/3d-slicer-extension). To edit topics, click the settings icon in the right side of "About" section header and enter `3d-slicer-extension` in "Topics" and click "Save changes". To learn more about topics, read https://help.github.com/en/articles/about-topics
- [x] Extension description summarizes in 1-2 sentences what the extension is usable (should be understandable for non-experts)
- [x] Any known related patents must be mentioned in the extension description.
- [x] If source code license is more restrictive for users than BSD, MIT, Apache, or 3D Slicer license then the name of the used license must be mentioned in the extension description.
- [x] Extension URL and revision (scmurl, scmrevision) is correct, consider using a branch name (master, release, ...) instead of a specific git has to avoid re-submitting pull request whenever the extension is updated
- [x] Extension icon URL is correct (do not use the icon's webpage but the raw data download URL that you get from the download button - it should look something like this: https://github.com/user/project/raw/main/path/to/icon.png)
- [x] Screenshot URLs (screenshoturls) are correct, contains at least one
- [x] Homepage URL points to valid webpage containing the following:
  - [x] Extension name
  - [x] Short description: 1-2 sentences, which summarizes what the extension is usable for
  - [x] At least one nice, informative image, that illustrates what the extension can do. It may be a screenshot.
  - [x] Description of contained modules: at one sentence for each module
  - [x] Tutorial: step-by-step description of at least the most typical use case, include a few screenshots, provide download links to sample input data set
  - [x] Publication: link to publication and/or to PubMed reference (if available)
  - [x] License: We suggest you use a permissive license that includes patent and contribution clauses.  This will help protect developers and ensure the code remains freely available.  We suggest you use the [Slicer License](https://github.com/Slicer/Slicer/blob/master/License.txt) or the [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0). Always mention in your README file the license you have chosen.  If you choose a different license, explain why to the extension maintainers. Depending on the license we may not be able to host your work. Read [here](https://opensource.guide/legal/#which-open-source-license-is-appropriate-for-my-project) to learn more about licenses.
  - [x] Content of submitted s4ext file is consistent with the top-level CMakeLists.txt file in the repository (description, URLs, dependencies, etc. are the same)

[If you have any questions or comments then please describe them here.]

pieper commented:
This is excellent! 👍 

A few comments:

* there's some dead code (commented out blocks) and it can be cleaner not to includes that
* do you have any sample data that you can share publicly? a cropped region with no scientific interest would be enough so that people could try the module out.  Ideally it could be bundled in a small module [like this one](https://github.com/SlicerMorph/SlicerMorph/tree/master/SlicerMorphSampleData) so it's easy to load.

lassoan commented:
You don't even have to create a new module but you can register sample data sets by adding a few lines of code to any of your existing module. See example [here](https://github.com/rbumm/SlicerLungCTAnalyzer/blob/a5700e941d7a2efe3b96a3d721dd4207bbdf12ac/LungCTAnalyzer/LungCTAnalyzer.py#L33-L75).

sindhurathiru commented:
Thank you very much for the feedback Steve and Andras, and sorry for the delay in getting back to you! For the sample data, I've added it to the GitHub page but haven't loaded it in as sample data for the module. For this data, the names of the folders containing each of the images is important and a hierarchy is needed (shown below), so I wasn't sure how to do that correctly as sample data. Hope that's okay! I've also removed the dead code.

![data hierarchy](https://user-images.githubusercontent.com/21988487/126254405-8c58801d-ec81-4150-ab60-f2db1cdfa153.PNG)

pieper commented:
That's fine - if the data doesn't fit the pattern of the SampleData module your workaround sounds good.
